### PR TITLE
Time in save games + Pausing Support

### PR DESCRIPTION
**Pull request by markusobi**
- game clock/day is now properly stored/loaded in save games
- prior save games will start at time=0 (they stored timePlayed=0)
- added version number to save games
- added game pausing when menus open/close
- saving/loading savegames closes menus new
- fixed not working pausekey after 2.world load
- pausing the game pauses all sound

### CODE REVIEW

File: src/engine/GameClock.h [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+         * helper function for conversion
+         * @param hours
+         * @param minutes
+         * @return converts timeOfDay in days (float) to hours/minutes
+         */
+        static void dayTimeTohm(float timeOfDay, int& hours, int& minutes);
+
+    private:
+        // Last deltatime-value we have gotten here
+        double lastFrameDeltaTime;
```
Comment (degenerated1123) [inline src/engine/GameClock.h]: Any reason for not using the `m_`-prefix for member variables here?

File: src/engine/World.cpp [hunk 2]
```diff
@@ -35,8 +35,6 @@ WorldInstance::WorldInstance()
       m_BspTree(*this),
       m_PfxManager(*this)
 {
-    // Both games start at 8:00 in the morning
-    m_WorldInfo.setTimeOfDay(8, 00);
```
Comment (degenerated1123) [inline src/engine/World.cpp]: Shouldn't put 00 here to not confuse people using the function for other times. Integer values with an extra 0 in front of them are seen as octal-numbers, thus everything over 7 would result in something different than intended.

Comment (degenerated1123) [inline src/engine/World.cpp]: Oh wait, that's removed code. Sorry!

File: src/engine/GameClock.h (hunk 1 above)
Comment (markusobi) [inline src/engine/GameClock.h]: I had a reason: I forgot the style standard. Will fix that soon.

**Pull request merged**